SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -403,6 +403,10 @@ bool USpatialLatencyTracer::BeginLatencyTrace_Internal(const FString& TraceDesc,
 		OutLatencyPayload = FSpatialLatencyPayload(MoveTemp(TraceBytes), MoveTemp(SpanBytes));
 	}
 
+	TraceKey Key = GenerateNewTraceKey();
+	TraceMap.Add(Key, MoveTemp(NewTrace));
+	ActiveTraceKey = Key; // This will be followed by ContinueLatencyTrace
```

Comment:
Shouldn't it be the responsibility of the user to pass the Trace back into the continue function? I feel we should move towards removing the `active` trace.

Reply:
I think there was a bug in the initial version because the root span is never closed currently, ContinueLatencyTrace does not end the previous span which would mean the root never got closed - this caused some weird looking traces where the root was swapped with the first Continue trace :/ I'll investigate improving the situation instead

Reply:
So I changed this code to just end the trace and then Continue will add it into the internal map, Sami has a doc for proposing better solutions for tracing (one of which includes removing the active trace concept)